lib/redis.js

```diff
@@ -95,11 +95,12 @@ module.exports = CoreObject.extend({
 
   _updateRecentUploadsList: function(keyPrefix, revisionKey) {
     var client = this._client;
-    return client.lpush(keyPrefix, revisionKey);
+    var score = new Date().getTime();
+    return client.zadd(keyPrefix, score, revisionKey);
   },
 
   _trimRecentUploadsList: function(keyPrefix, maxEntries) {
     var client = this._client;
-    return client.ltrim(keyPrefix, 0, maxEntries - 1);
+    return client.zremrangebyrank(keyPrefix, 0, -(maxEntries + 1));
```

Comment:
I'd like to see this addressed in this PR. I think it will make a lot more sense to readers as a single unit.